### ISSUE TITLE
Multithread - Access violation when a json is deserialized

### ISSUE THREAD

**Issue by daviladanielc** [paolo-rossi/delphi-neon#77]
I'm experiencing an access violation issue when there is a high level of concurrency with multithreading. I have a service that receives JSON data to insert a record into a database table. I'm using Neon to deserialize the JSON and map it to a class.

When I receive a request in my service, I handle it like this:

```
TTask.Run(
  procedure
  begin
    TPersistencia.Insert(StompFrame.Body, FConf, StompFrame, Self.FStompClient);
  end
);
```

In my tests, I sent the exact same JSON 1000 times, and occasionally, during the execution of the process, an access violation occurs when TNeonDeserializerJSON attempts to read the JSON to populate the class:

```
LReader := TNeonDeserializerJSON.Create(LConfig);
LReader.JSONToObject(LClasseTabela, LJSON);  //<< AV occurs here
```

If I remove TTask, it works fine but is slower, and I need it to be faster. How can I address this problem?

Here is the callstacks of Delphi:
![image](https://github.com/paolo-rossi/delphi-neon/assets/71415962/b655f352-8bda-441b-8558-e0d41ac96d50)

**Comment by daviladanielc** [comment]
I tried using TMonitor within a TCriticalSection when LReader is executed, but the problem persists.

**Comment by paolo-rossi** [comment]
Hello @daviladanielc ,

thank you for taking the time to file this issue...

Can build a small test case that reproduce the issue?  thanks!

Paolo.

**Comment by paolo-rossi** [comment]
No additional info, probably it's been fixed along issue #83. Closing.
